### PR TITLE
Fix commit's time

### PR DESCRIPTION
**Pull request by Gusted**
- Use the Committer's `when` as the CommitDate will be modified when the commit is being moved around in the git tree(e.g. being rebased on top of newer commits). The Author's `when` is created once and never touched again.
- TL;DR It will use the committed time(which is "dynamic") instead of the creation time(which is "static")

Before(The top 3 commits were rebased on top of the last 2 commits):
image

After:
image

Github's commit list:
image

### CODE REVIEW

**Comment by zeripath** [comment]
What does Github do here? I'm fairly certain it uses Committer time.

**Comment by zeripath** [comment]
git log will use date-order as in committer time and we have to use committer time for our last commit time for commits too.

File: templates/repo/view_list.tmpl [hunk 1]
```diff
@@ -34,7 +34,7 @@
 					</span>
 				{{end}}
 			</th>
-			<th class="text grey right age">{{if .LatestCommit}}{{if .LatestCommit.Author}}{{TimeSince .LatestCommit.Author.When $.Lang}}{{end}}{{end}}</th>
+			<th class="text grey right age">{{if .LatestCommit}}{{if .LatestCommit.Author}}{{TimeSince .LatestCommit.Committer.When $.Lang}}{{end}}{{end}}</th>
```
Comment (zeripath) [inline templates/repo/view_list.tmpl]:
```suggestion
			<th class="text grey right age">{{if .LatestCommit}}{{if .LatestCommit.Committer}}{{TimeSince .LatestCommit.Committer.When $.Lang}}{{end}}{{end}}</th>
```

**Review by zeripath — APPROVED** [review]

**Comment by Gusted** [comment]
> What does Github do here? I'm fairly certain it uses Committer time.

Yup it does, this PR switch to that behaviors. It's only pure visual.

> git log will use date-order as in committer time and we have to use committer time for our last commit time for commits too.

Will update the PR to include that(if it doesn't already)

**Comment by zeripath** [comment]
> > What does Github do here? I'm fairly certain it uses Committer time.
> 
> Yup it does, this PR switch to that behaviors. It's only pure visual.
> 
> > git log will use date-order as in committer time and we have to use committer time for our last commit time for commits too.
> 
> Will update the PR to include that(if it doesn't already)

Yeah it should always have been the committer date I was confused when I saw your PR because I thought you were changing it to Author not Committer!

**Comment by Gusted** [comment]
Dammit, the latest commit is the actual PR(as in, what you see on the screenshots) the first commit is only for the `latest commit` bar when you view a repo.

**Review by KN4CK3R — APPROVED** [review]

**Comment by lunny** [comment]
`Commiter` maybe nil. We need a fallback to `Author`

**Comment by silverwind** [comment]
This should solve the order of commits on https://try.gitea.io/silverwind/symlink-test/commits/branch/master, right?

image

**Comment by Gusted** [comment]
> This should solve the order of commits on https://try.gitea.io/silverwind/symlink-test/commits/branch/master, right?

From what I can see, it should be.

**Comment by Gusted** [comment]
> `Commiter` maybe nil. We need a fallback to `Author`

Fallback has been added.

**Review by lafriks — APPROVED** [review]

**Comment by silverwind** [comment]
Can confirm, it fixed https://try.gitea.io/silverwind/symlink-test/commits/branch/master.

**Comment by lunny** [comment]
@Gusted When sent backport, the lable `backport/v1.16` should be still kept so that we can filter them.

And I think we should also backport this to v1.15.11 .

**Pull request merged**